tests/test_coordinated_workers/test_coordinator.py

```diff
@@ -344,3 +347,72 @@ def test_invalid_databag_content(coordinator_charm: ops.CharmBase, event):
         cluster.gather_addresses_by_role()
         manager.run()
     assert cluster.model.unit.status == ops.BlockedStatus("[consistency] Cluster inconsistent.")
+
+
+@patch('cosl.coordinated_workers.interface.ClusterProviderAppData.load',
+       side_effect=DataValidationError("Mock error"))
+def test_invalid_coordinator_databag(mock_databag_load, coordinator_charm: ops.CharmBase, coordinator_state):
+    # Test that when a relation changes and the relation app data is invalid
+    #   the worker emits a ClusterRemovedEvent
+
+    # WHEN you define a properly configured charm
+    ctx = testing.Context(
+        MyCharm,
+        meta={
+            "name": "foo",
+            "requires": {"cluster": {"interface": "cluster"}},
+            "containers": {"foo": {"type": "oci-image"}},
+        },
+        config={"options": {"role-all": {"type": "boolean", "default": True}}},
+    )
+
+    # AND the relation changes
+    # IF the relation data is invalid (forced by the patched Exception)
+    relation = testing.Relation(
+        "cluster",
+        remote_app_data={"worker_config": json.dumps("some: yaml")},
```

Comment:
I believe that this line is unnecessary, since the cluster provider app data that you are checking is the local side of the relation (from the pov of the coordinator).
Also, worker_config is a field in the coordinator's databag, so it should be on 'this side'.

Reply:
Removed in [0005a0b](https://github.com/canonical/cos-lib/pull/107/commits/0005a0b82548eed2448961c87bd4dd4f29bc0486).